cmd/node/main.go

```diff
@@ -14,95 +15,80 @@ import (
-			api.AsRoute(controller.NewGetNodeRoute),
-			api.AsRoute(controller.NewAddUserRoute),
-			api.AsRoute(controller.NewInstallAppRoute),
-			api.AsRoute(controller.NewStartProcessHandler),
-			api.AsRoute(controller.NewMigrationRoute),
-		),
-		fx.Provide(
-			reverse_proxy.NewProxyServer,
-		),
-		fx.Provide(
-			fx.Annotate(
-				docker.NewDockerDriver,
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				hdbms.NewStateUpdateLogger,
-				fx.As(new(pubsub.Subscriber[hdb.StateUpdate])),
-				fx.ResultTags(`group:"state_update_subscribers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				package_manager.NewAppLifecycleSubscriber,
-				fx.As(new(pubsub.Subscriber[hdb.StateUpdate])),
-				fx.ResultTags(`group:"state_update_subscribers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				processes.NewProcessManagerStateUpdateSubscriber,
-				fx.As(new(pubsub.Subscriber[hdb.StateUpdate])),
-				fx.ResultTags(`group:"state_update_subscribers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				reverse_proxy.NewProcessProxyRuleStateUpdateSubscriber,
-				fx.As(new(pubsub.Subscriber[hdb.StateUpdate])),
-				fx.ResultTags(`group:"state_update_subscribers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				processes.NewProcessManager,
-				fx.ParamTags(`group:"process_drivers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				pubsub.NewSimpleChannel[hdb.StateUpdate],
-				fx.As(new(pubsub.Channel[hdb.StateUpdate])),
-				fx.ParamTags(`group:"state_update_publishers"`, `group:"state_update_subscribers"`),
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				hdbms.NewHabitatDB,
-			),
-		),
-		fx.Provide(
-			fx.Annotate(
-				controller.NewNodeController,
-				fx.As(new(controller.NodeController)),
-			),
-		),
-		fx.Invoke(func(controller.NodeController) {}),
-		fx.Invoke(func(*http.Server) {}),
-		fx.Invoke(func(pubsub.Channel[hdb.StateUpdate]) {}),
-	).Run()
+	ctx := context.Background()
+	log := logging.NewLogger()
+
+	nodeConfig, err := config.NewNodeConfig()
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	hdbPublisher := pubsub.NewSimplePublisher[hdb.StateUpdate]()
+	db, dbClose, err := hdbms.NewHabitatDB(log, nodeConfig)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+	defer dbClose()
+
+	nodeCtrl, err := controller.NewNodeController(db.Manager, nodeConfig)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	routes := []api.Route{
+		api.NewVersionHandler(),
+		controller.NewGetNodeRoute(db.Manager),
+		controller.NewAddUserRoute(nodeCtrl),
+		controller.NewInstallAppRoute(nodeCtrl),
+		controller.NewStartProcessHandler(nodeCtrl),
+		controller.NewMigrationRoute(nodeCtrl),
+	}
+
+	router := api.NewRouter(routes, log, nodeCtrl, nodeConfig)
+	proxy := reverse_proxy.NewProxyServer(log, nodeConfig)
+	tlsConfig, err := nodeConfig.TLSConfig()
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+	proxyClose, err := proxy.Start(ctx, constants.DefaultPortReverseProxy, tlsConfig)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+	defer proxyClose()
```

Comment:
Hm I looked this up and it said no ....
I think what we want is:
trap the SIGINT -> graceful shutdown (cancel top-level context that all downstreams are listening on)
if any other signal happens -> forcefully close

I'm down to do this in a separate PR since right now all the servers just forcefully call `Close()` rather than the `Shutdown(ctx)` , don't wanna block the other open PRs
Wanna go back through and do a deeper quality pass